packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2106,14 +2097,11 @@ def process_event(
         # we first check whether the event is the special Termination event.
         # if that's the case, we move to `transaction_settlement_abci`,
         # regardless of what the current round is
-        if event == Event.TERMINATE:
-            next_round_cls = self._get_transaction_settlement_initial_round()
-            if next_round_cls is None:
-                raise ValueError(
-                    "Termination majority was reached, "
-                    "but the transition to `transaction_settlement_abci` failed."
-                )
-
+        if self._is_termination_set and event == self.termination_event:
+            next_round_cls = self.termination_transition_function[self.termination_round_cls].get(
+                event, None
+            )
+            self.transition_function = self.termination_transition_function
```

Comment:
Add a comment in code here that entering the termination is a one-way operation.